Objects/abstract.c

```diff
@@ -819,6 +819,20 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (op_slot == NB_SLOT(nb_rshift) && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
```

Comment:
Right, that was my reasoning too - while we *could* make the actual symbol available here, "It's a function called 'print' implemented in C" is a robust enough check, and it means the C level changes can stay contained to this module.